catalog/core/catalog-core-commands/src/main/java/org/codice/ddf/commands/catalog/SeedCommand.java

```diff
@@ -0,0 +1,241 @@
+import org.apache.karaf.shell.api.action.Option;
+import org.apache.karaf.shell.api.action.lifecycle.Reference;
+import org.apache.karaf.shell.api.action.lifecycle.Service;
+import org.geotools.filter.text.cql2.CQL;
+import org.opengis.filter.Filter;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import ddf.catalog.CatalogFramework;
+import ddf.catalog.data.Attribute;
+import ddf.catalog.data.Metacard;
+import ddf.catalog.data.Result;
+import ddf.catalog.filter.FilterBuilder;
+import ddf.catalog.operation.QueryResponse;
+import ddf.catalog.operation.ResourceRequest;
+import ddf.catalog.operation.ResourceResponse;
+import ddf.catalog.operation.impl.QueryImpl;
+import ddf.catalog.operation.impl.QueryRequestImpl;
+import ddf.catalog.operation.impl.ResourceRequestById;
+import ddf.catalog.resource.ResourceNotFoundException;
+import ddf.catalog.resource.ResourceNotSupportedException;
+
+//@formatter:off
+@Service
+@Command(
+        scope = CatalogCommands.NAMESPACE,
+        name = "seed",
+        description = "Seeds the local metacard and resource caches from the enterprise or from specific federated sources.",
+        detailedDescription = "This command will query the enterprise or the specified sources for metacards in increments "
+                + "the size of the `--resource-limit` argument (default 20) until that number of resource downloads "
+                + "have started or until there are no more results. Local resources will not be added to the cache. "
+                + "This command will not re-download resources that are up-to-date in the cache, so subsequent runs "
+                + "of the command will attempt to cache metacards and resources that have not already been cached.\n\n"
+                + "Note: this command will trigger resource downloads in the background and they may continue after "
+                + "control is returned to the console. Also, resource caching must be enabled in the catalog framework "
+                + "for this command to seed the resource cache.")
+//@formatter:on
+public class SeedCommand extends SubjectCommands implements Action {
+    private static final Logger LOGGER = LoggerFactory.getLogger(SeedCommand.class);
+
+    private static final Map<String, Serializable> CACHE_UPDATE_PROPERTIES =
+            Collections.singletonMap("mode", "update");
+
+    private static final String RESOURCE_CACHE_STATUS = "internal.local-resource";
+
+    private static final ThreadPoolExecutor EXECUTOR;
+
+    static {
+        EXECUTOR = new ThreadPoolExecutor(20,
+                20,
+                30,
+                TimeUnit.SECONDS,
+                new LinkedBlockingQueue<>());
+        EXECUTOR.allowCoreThreadTimeOut(true);
+    }
+
+    @Option(name = "--source", multiValued = true, aliases = {
+            "-s"}, description = "Source(s) to query (e.g., -s source1 -s source2). Default is to query the entire enterprise.")
+    List<String> sources;
```

Comment:
The Karaf documentation says the field must be a `List`: https://github.com/apache/karaf/blob/master/shell/core/src/main/java/org/apache/karaf/shell/api/action/Option.java#L86